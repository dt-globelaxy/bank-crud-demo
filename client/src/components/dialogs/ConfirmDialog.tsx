import * as React from 'react';
import Dialog from '@material-ui/core/Dialog/Dialog';
import DialogTitle from '@material-ui/core/DialogTitle/DialogTitle';
import DialogContent from '@material-ui/core/DialogContent/DialogContent';
import DialogActions from '@material-ui/core/DialogActions/DialogActions';
import Button from '@material-ui/core/Button/Button';
import Typography from '@material-ui/core/Typography/Typography';

interface IConfirmDialogProps {
    open: boolean;
    title?: string;
    message: string;
    yesButtonText?: string;
    noButtonText?: string;
    handleDialogClose: Function;
    handleDialogAction: Function;
}

class ConfirmDialog extends React.Component<IConfirmDialogProps> {
    render() {
        return (
        <Dialog open={this.props.open} onClose={() => this.props.handleDialogClose()}>
            <DialogTitle>{this.props.title ? this.props.title : 'Confirmation'}</DialogTitle>
            <DialogContent>
                <Typography>{this.props.message ? this.props.message : 'Do you confirm action?'}</Typography>
            </DialogContent>
            <DialogActions>
              <Button color="primary" onClick={() => this.props.handleDialogClose()}>
                {this.props.noButtonText ? this.props.noButtonText : 'No'}
              </Button>
              <Button color="primary" onClick={() => this.props.handleDialogAction()}>
                {this.props.yesButtonText ? this.props.yesButtonText : 'Yes'}
              </Button>
            </DialogActions>
        </Dialog>)
    }
}

export default ConfirmDialog;