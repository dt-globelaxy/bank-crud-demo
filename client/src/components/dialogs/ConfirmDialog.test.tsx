import React from 'react';
import ConfirmDialog from './ConfirmDialog';
import Button from '@material-ui/core/Button/Button';
import { createShallow, createMount, createRender } from '@material-ui/core/test-utils';
import { Typography, DialogTitle } from '@material-ui/core';

describe('ConfirmDialog component', () => {
    let shallow: any;
    let mount: any;
    let render: any;

    beforeAll(() => {
      shallow = createShallow();
      mount = createMount();
      render = createRender();
    });

    it('renders without crashing', () => {
        const mockHandleDialogClose = jest.fn();
        const mockHandleDialogAction = jest.fn();
        const wrapper = shallow(<ConfirmDialog 
            open={true} 
            message={'Test message'} 
            handleDialogClose={mockHandleDialogClose} 
            handleDialogAction={mockHandleDialogAction}
        />);
        expect(wrapper).toMatchSnapshot();
        wrapper.find(Button).first().simulate('click');
        expect(mockHandleDialogClose.mock.calls.length).toEqual(1);
        wrapper.find(Button).last().simulate('click');
        expect(mockHandleDialogAction.mock.calls.length).toEqual(1);
        wrapper.simulate('close');
        expect(mockHandleDialogClose.mock.calls.length).toEqual(2);
    });

    it('renders message and title', () => {
        const mockHandleDialogClose = jest.fn();
        const mockHandleDialogAction = jest.fn();
        const wrapper = shallow(<ConfirmDialog 
            open={false} 
            title={'Test title'} 
            message={'Test message'} 
            handleDialogClose={mockHandleDialogClose} 
            handleDialogAction={mockHandleDialogAction}
        />);
        expect(wrapper.find(Typography).html()).toMatch(/Test message/);
        expect(wrapper.find(DialogTitle).html()).toMatch(/Test title/);
    });

    it('renders default message if param is empty', () => {
        const mockHandleDialogClose = jest.fn();
        const mockHandleDialogAction = jest.fn();
        const wrapper = shallow(<ConfirmDialog 
            open={false} 
            message={''} 
            handleDialogClose={mockHandleDialogClose} 
            handleDialogAction={mockHandleDialogAction}
        />);
        expect(wrapper.find(Typography).html()).toMatch(/Do you confirm action\?/);
    });

    it('renders default values in title and Yes No buttons', () => {
        const mockHandleDialogClose = jest.fn();
        const mockHandleDialogAction = jest.fn();
        const wrapper = shallow(<ConfirmDialog 
            open={false} 
            message={'Test message'} 
            handleDialogClose={mockHandleDialogClose} 
            handleDialogAction={mockHandleDialogAction}
        />);
        expect(wrapper.find(DialogTitle).html()).toMatch(/Confirmation/);
        expect(wrapper.find(Button).first().html()).toMatch(/No/);
        expect(wrapper.find(Button).last().html()).toMatch(/Yes/);
    });

    it('renders Yes and No buttons', () => {
        const mockHandleDialogClose = jest.fn();
        const mockHandleDialogAction = jest.fn();
        const wrapper = shallow(<ConfirmDialog 
            open={false} 
            message={'Test message'} 
            yesButtonText={'Custom Yes'}
            noButtonText={'Custom No'}
            handleDialogClose={mockHandleDialogClose} 
            handleDialogAction={mockHandleDialogAction}
        />);
        expect(wrapper.find(Button).first().html()).toMatch(/Custom No/);
        expect(wrapper.find(Button).last().html()).toMatch(/Custom Yes/);
    });
});