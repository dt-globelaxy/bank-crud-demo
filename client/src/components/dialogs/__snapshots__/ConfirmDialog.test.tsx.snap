// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmDialog component renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ConfirmDialog
    handleDialogAction={[MockFunction]}
    handleDialogClose={[MockFunction]}
    message="Test message"
    open={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <WithStyles(DialogTitle)>
          Confirmation
        </WithStyles(DialogTitle)>,
        <WithStyles(DialogContent)>
          <WithStyles(Typography)>
            Test message
          </WithStyles(Typography)>
        </WithStyles(DialogContent)>,
        <WithStyles(DialogActions)>
          <WithStyles(Button)
            color="primary"
            onClick={[Function]}
          >
            No
          </WithStyles(Button)>
          <WithStyles(Button)
            color="primary"
            onClick={[Function]}
          >
            Yes
          </WithStyles(Button)>
        </WithStyles(DialogActions)>,
      ],
      "onClose": [Function],
      "open": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Confirmation",
        },
        "ref": null,
        "rendered": "Confirmation",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(Typography)>
            Test message
          </WithStyles(Typography)>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Test message",
          },
          "ref": null,
          "rendered": "Test message",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(Button)
              color="primary"
              onClick={[Function]}
            >
              No
            </WithStyles(Button)>,
            <WithStyles(Button)
              color="primary"
              onClick={[Function]}
            >
              Yes
            </WithStyles(Button)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "No",
              "color": "primary",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "No",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Yes",
              "color": "primary",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Yes",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(DialogTitle)>
            Confirmation
          </WithStyles(DialogTitle)>,
          <WithStyles(DialogContent)>
            <WithStyles(Typography)>
              Test message
            </WithStyles(Typography)>
          </WithStyles(DialogContent)>,
          <WithStyles(DialogActions)>
            <WithStyles(Button)
              color="primary"
              onClick={[Function]}
            >
              No
            </WithStyles(Button)>
            <WithStyles(Button)
              color="primary"
              onClick={[Function]}
            >
              Yes
            </WithStyles(Button)>
          </WithStyles(DialogActions)>,
        ],
        "onClose": [Function],
        "open": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Confirmation",
          },
          "ref": null,
          "rendered": "Confirmation",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Typography)>
              Test message
            </WithStyles(Typography)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Test message",
            },
            "ref": null,
            "rendered": "Test message",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(Button)
                color="primary"
                onClick={[Function]}
              >
                No
              </WithStyles(Button)>,
              <WithStyles(Button)
                color="primary"
                onClick={[Function]}
              >
                Yes
              </WithStyles(Button)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "No",
                "color": "primary",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "No",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Yes",
                "color": "primary",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Yes",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {},
  },
  Symbol(enzyme.__childContext__): null,
}
`;
